pkg/framework/report.go

```diff
@@ -132,14 +147,27 @@ func parsePodsReview(templatePods []*api.Pod, status Status) []*ClusterCapacityR
 
 	for i := 0; i < templatesCount; i++ {
 		result = append(result, &ClusterCapacityReviewResult{
-			ReplicasOnNodes: make(map[string]int),
+			ReplicasOnNodes: make([]ReplicasOnNode, 0),
 			PodName:         templatePods[i].Name,
 		})
 	}
 
+	replicaCount := make([]map[string]int,templatesCount)
 	for i, pod := range status.Pods {
 		nodeName := pod.Spec.NodeName
-		result[i%templatesCount].ReplicasOnNodes[nodeName]++
+		if replicaCount[i%templatesCount] == nil {
+			replicaCount[i%templatesCount] = make(map[string]int)
+		}
+		replicaCount[i%templatesCount][nodeName]++
+	}
+
+	for i, val := range replicaCount {
+		for k, v := range val {
+			result[i].ReplicasOnNodes = append(result[i].ReplicasOnNodes, ReplicasOnNode{
+				NodeName: k,
+				Replicas: v,
+			})
+		}
```

Comment:
I think this could be avoided if you just used ReplicasOnNode above instead of the map that you are anyway getting rid of in this PR.